test/replcompletions.jl

```diff
@@ -494,6 +493,29 @@ c, r, res = test_scomplete(s)
         ENV["PATH"] = oldpath
     end
 
+    # Make sure completion results are unique in case things are in the env path twice.
+    let
+        # Create /tmp/repl-completion and /tmp/repl-completion-subdir/repl-completion
+        file0 = joinpath(tempdir(), "repl-completion")
+        dir = joinpath(tempdir(), "repl-completion-subdir")
+        mkdir(dir)
+        file1 = joinpath(dir, "repl-completion")
+        touch(file0)
+        touch(file1)
+
+        oldpath = ENV["PATH"]
+        ENV["PATH"] = string(tempdir(), ":", dir)
```

Comment:
Changed to use `withenv` now.

I plan to make all the (messy) tests I wrote run on Windows, because I didn't do that initially as I didn't have a Windows installation handy. I'll try to set up a Windows VM and get that done during the weekend (as a separate PR, then).